spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/HealthCheckServiceInstanceListSupplier.java

```diff
@@ -40,10 +41,11 @@
  * {@link WebClient} to ping the <code>health</code> endpoint of the instances.
  *
  * @author Olga Maciaszek-Sharma
+ * @author Roman Matiushchenko
  * @since 2.2.0
  */
 public class HealthCheckServiceInstanceListSupplier
-		implements ServiceInstanceListSupplier {
+		implements ServiceInstanceListSupplier, DisposableBean {
```

Comment:
I've tried to make minimal changes to the current logic but    
I've found testing of ping functionality a little bit inconvenient  
`HealthCheckServiceInstanceListSupplier` starts ping implicitly and eventually I test just another stream which uses common state...     
IMO It signals that this functionality must be either moved to a new class or just spitted into different methods.  
So I would like to make `HealthCheckServiceInstanceListSupplier` an `InitializingBean` , and move start of pings out of constructor   
and  
make this  
```java
Flux<List<ServiceInstance>> instancesFlux = this.delegate.get().doOnNext(delegateInstances -> {
	this.instances = Collections.unmodifiableList(new ArrayList<>(delegateInstances));
});
```
and 
```java 
Flux<List<ServiceInstance>> verifiedInstancesFlux = healthCheckFlux()
		.doOnNext((verifiedInstances -> {
			this.healthyInstances = verifiedInstances;
		}));
```
two distinct methods which are invoked from  `afterPropertiesSet()`  
does this make sense @spencergibb  @OlgaMaciaszek